pkg/querier/frontend/frontend.go

```diff
@@ -426,3 +433,23 @@ FindQueue:
 	// and wait for more requests.
 	goto FindQueue
 }
+
+// CheckReady determines if the query frontend is ready.  Function parameters/return
+// chosen to match the same method in the ingester
+func (f *Frontend) CheckReady(_ context.Context) error {
+	// if the downstream url is configured the query frontend is not aware of the state
+	//  of the queriers and is therefore always ready
+	if f.cfg.DownstreamURL != "" {
+		return nil
+	}
+
+	// if we have more than one querier connected we will consider ourselves ready
+	connectedClients := atomic.LoadInt32(&f.connectedClients)
+	if connectedClients > 0 {
+		return nil
+	}
+
+	msg := fmt.Sprintf("not ready: connected clients %d", connectedClients)
```

Comment:
right, i included this detail in case we were unexpected receiving service unavailable.  i don't believe it's possible for connectedClients to be negative, but you know how that goes in complex systems :)